openmdao/core/system.py

```diff
@@ -2661,11 +2676,41 @@ def add_response(self, name, type_, lower=None, upper=None, equals=None,
             except ValueError as e:
                 raise(ValueError(f"{str(e)[:-1]} for response '{name}'."))
 
+        resp = OrderedDict()
+
         typemap = {'con': 'Constraint', 'obj': 'Objective'}
-        if name in self._responses or name in self._static_responses:
-            msg = "{}: {} '{}' already exists.".format(self.msginfo, typemap[type_], name)
+        if (name in self._responses or name in self._static_responses) and alias is None:
+            msg = ("{}: {} '{}' already exists. Use the 'alias' argument to apply a second "
+                   "constraint".format(self.msginfo, typemap[type_], name))
             raise RuntimeError(msg.format(name))
 
+        if alias is not None:
+            resp['name'] = alias
+            resp['path'] = name
+        else:
+            resp['name'] = name
+            resp['path'] = None
+
+        self._multi_constraint[name] = False
+
+        if indices is not None:
+            if name in self._resp_indices_map:
+                if resp['path'] is not None:
+                    msg = (f"{self.msginfo}: '{resp['name']}' indices are overlapping its parent "
```

Comment:
The choice of words here is confusing.  The conflicting responses don't have a parent/child relationship.